libafl/src/corpus/testcase.rs

```diff
@@ -144,8 +146,53 @@ where
 
     /// Set the filename
     #[inline]
-    pub fn set_filename(&mut self, filename: String) {
+    #[cfg(feature = "std")]
+    pub fn set_filename(&mut self, filename: String) -> Result<(), Error> {
+        use std::fs::OpenOptions;
+
+        if self.filename.is_some() {
+            let f = self.filename.clone().unwrap();
```

Comment:
I think it's okay to `take()` the old `filename` here, then again we don't need an extra allocation.
Also, it's a good idea to first check if the new filename is different from the old one, else do nothing :)